nvme.c

```diff
@@ -5317,6 +5321,289 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
+		  OPT_FLAG("cmbsz",          'M', &cfg.cmbsz, cmbsz),
+		  OPT_FLAG("bpinfo",         'b', &cfg.bpinfo, bpinfo),
+		  OPT_FLAG("bprsel",         'B', &cfg.bprsel, bprsel),
+		  OPT_FLAG("bpmbl",          'p', &cfg.bpmbl, bpmbl),
+		  OPT_FLAG("cmbmsc",         's', &cfg.cmbmsc, cmbmsc),
+		  OPT_FLAG("cmbsts",         'S', &cfg.cmbsts, cmbsts),
+		  OPT_FLAG("crto",           'r', &cfg.crto, crto),
+		  OPT_FLAG("pmrcap",         'P', &cfg.pmrcap, pmrcap),
+		  OPT_FLAG("pmrctl",         'R', &cfg.pmrctl, pmrctl),
+		  OPT_FLAG("pmrsts",         't', &cfg.pmrsts, pmrsts),
+		  OPT_FLAG("pmrebs",         'e', &cfg.pmrebs, pmrebs),
+		  OPT_FLAG("pmrswtp",        'w', &cfg.pmrswtp, pmrswtp),
+		  OPT_FLAG("pmrmscl",        'l', &cfg.pmrmscl, pmrmscl),
+		  OPT_FLAG("pmrmscu",        'u', &cfg.pmrmscu, pmrmscu));
+
+	err = parse_and_open(&dev, argc, argv, desc, opts);
+	if (err)
+		return err;
+
+	if (!argconfig_parse_seen(opts, "offset")) {
+		if (argconfig_parse_seen(opts, "cap"))
+			cfg.offset = NVME_REG_CAP;
+		else if (argconfig_parse_seen(opts, "vs"))
+			cfg.offset = NVME_REG_VS;
+		else if (argconfig_parse_seen(opts, "intms"))
+			cfg.offset = NVME_REG_INTMS;
+		else if (argconfig_parse_seen(opts, "intmc"))
+			cfg.offset = NVME_REG_INTMC;
+		else if (argconfig_parse_seen(opts, "cc"))
+			cfg.offset = NVME_REG_CC;
+		else if (argconfig_parse_seen(opts, "csts"))
+			cfg.offset = NVME_REG_CSTS;
+		else if (argconfig_parse_seen(opts, "nssr"))
+			cfg.offset = NVME_REG_NSSR;
+		else if (argconfig_parse_seen(opts, "aqa"))
+			cfg.offset = NVME_REG_AQA;
+		else if (argconfig_parse_seen(opts, "asq"))
+			cfg.offset = NVME_REG_ASQ;
+		else if (argconfig_parse_seen(opts, "acq"))
+			cfg.offset = NVME_REG_ACQ;
+		else if (argconfig_parse_seen(opts, "cmbloc"))
+			cfg.offset = NVME_REG_CMBLOC;
+		else if (argconfig_parse_seen(opts, "cmbsz"))
+			cfg.offset = NVME_REG_CMBSZ;
+		else if (argconfig_parse_seen(opts, "bpinfo"))
+			cfg.offset = NVME_REG_BPINFO;
+		else if (argconfig_parse_seen(opts, "bprsel"))
+			cfg.offset = NVME_REG_BPRSEL;
+		else if (argconfig_parse_seen(opts, "bpmbl"))
+			cfg.offset = NVME_REG_BPMBL;
+		else if (argconfig_parse_seen(opts, "cmbmsc"))
+			cfg.offset = NVME_REG_CMBMSC;
+		else if (argconfig_parse_seen(opts, "cmbsts"))
+			cfg.offset = NVME_REG_CMBSTS;
+		else if (argconfig_parse_seen(opts, "crto"))
+			cfg.offset = NVME_REG_CRTO;
+		else if (argconfig_parse_seen(opts, "pmrcap"))
+			cfg.offset = NVME_REG_PMRCAP;
+		else if (argconfig_parse_seen(opts, "pmrctl"))
+			cfg.offset = NVME_REG_PMRCTL;
+		else if (argconfig_parse_seen(opts, "pmrsts"))
+			cfg.offset = NVME_REG_PMRSTS;
+		else if (argconfig_parse_seen(opts, "pmrebs"))
+			cfg.offset = NVME_REG_PMREBS;
+		else if (argconfig_parse_seen(opts, "pmrswtp"))
+			cfg.offset = NVME_REG_PMRSWTP;
+		else if (argconfig_parse_seen(opts, "pmrmscl"))
+			cfg.offset = NVME_REG_PMRMSCL;
+		else if (argconfig_parse_seen(opts, "pmrmscu"))
+			cfg.offset = NVME_REG_PMRMSCU;
+	}
```

Comment:
Corrected by the commit 206f9399.

> Also, should there be checks to enforce mutual exclusivity of these arguments?

I do not think so since nvme-cli not multi threads.

Reply:
> nvme-cli not multi threads.

I don't understand this response. I'm not talking at all about multithreading. I just mean that it would be invalid, for example, to specify `--offset 0 --vs` or `--cap --cc`, but the current code will silently ignore the second argument. The options to select a register should be mutually exclusive, i.e. at most one of them should be allowed.

Reply:
Sorry misunderstood but now got it so will consider to fix the issue later. Thank you for your advice. By the way I also thought this needed at first but forgot it to do.

Reply:
The current version (2024-03-25) still has this problem with two nested loops for checking if the `--offset` and `--regname` is used twice. Hmm, maybe we should just state, either user `--offset` OR `--regname`, not both at the same time.  This would make all this rather complex code checker unnecessary.

Reply:
Sorry let me confirm the comment. Currently the `--regnam` is checked if duplicated with the `--offset` by the function `set_register_names_check()` called in `set_register_offset()` so no problem I think but still the issue as mentioned?